api/src/main/java/org/openmrs/module/legacyui/EncounterEventListener.java

```diff
@@ -0,0 +1,202 @@
+	
+	private static final String DEATH_DATE_CONCEPT_UUID = "165233AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA";
+	
+	private DaemonToken daemonToken;
+	
+	@Autowired
+	DbSessionFactory sessionFactory;
+	
+	private DbSession getSession() {
+		return sessionFactory.getCurrentSession();
+	}
+	
+	public EncounterEventListener(DaemonToken token) {
+		daemonToken = token;
+	}
+	
+	public EncounterEventListener() {
+	}
+	
+	public DaemonToken getDaemonToken() {
+		return daemonToken;
+	}
+	
+	public void setDaemonToken(DaemonToken daemonToken) {
+		this.daemonToken = daemonToken;
+	}
+	
+	@Override
+	public void onMessage(Message message) {
+		log.trace(String.format("Received message: \n%s", message));
+		System.out.println("IN EVENT LISTENER ::: EncounterEventListener");
+		try {
+			Daemon.runInDaemonThread(() -> {
+				try {
+					processMessage(message);
+				}
+				catch (Exception e) {
+					log.error(String.format("Failed to process obs message!\n%s", message.toString()), e);
+				}
+			}, daemonToken);
+		} catch (Exception e) {
+			log.error(String.format("Failed to start Daemon thread to process message!\n%s", message.toString()), e);
+		}
+
+	}
+	
+	private void processMessage(Message message) throws Exception {
+		
+		MapMessage mapMessage = (MapMessage) message;
+		
+		String uuid = mapMessage.getString("uuid");
+		
+		Encounter encounter = Context.getEncounterService().getEncounterByUuid(uuid);
+		
+		String opencrMatchesCheckFlag = Context.getAdministrationService().getGlobalProperty("legacyui.enableMatchCheck",
+		    "true");
+		
+		// EXIT PATIENT ENCOUTER 
+		if (encounter.getEncounterType().getEncounterTypeId().compareTo(EXIT_PROGRAM_ENCOUNTER_TYPE_ID) == 0
+		        && opencrMatchesCheckFlag.equals("true")) {
+			
+			Patient patient = encounter.getPatient();
+			Set<Obs> obsSet = encounter.getObs();
+			
+			for (Obs obs : encounter.getObs()) {
+				
+				// TRANSFER
+				if (obs.getConcept().getUuid().compareTo(TRANSFERT_FORM_CONCEPT_ID) == 0) {
+					
+					Date tranfertDate = getObsRelatedDate(obsSet, TRANSFERT_DATE_CONCEPT_UUID);
+					updatePatientAttributes(patient, TRANSFERED_CONCEPT_VALUE_ID, tranfertDate);
+					break;
+				}
+				
+				// VOLUNTARY STOP
+				if (obs.getConcept().getUuid().compareTo(SELFSTOPPED_FORM_CONCEPT_ID) == 0) {
+					
+					Date stopDate = getObsRelatedDate(obsSet, STOP_DATE_CONCEPT_UUID);
+					updatePatientAttributes(patient, STOP_CONCEPT_VALUE_ID, stopDate);
+					break;
+				}
+				
+				// DEATH
+				if (obs.getConcept().getUuid().compareTo(DECEASED_FORM_CONCEPT_ID) == 0) {
+					
+					Date deathDate = getObsRelatedDate(obsSet, DEATH_DATE_CONCEPT_UUID);
+					updatePatientAttributes(patient, DEATH_CONCEPT_VALUE_ID, deathDate);
+					break;
+				}
+				
+				// NEGATIF
+				if (obs.getConcept().getUuid().compareTo(NEGATIF_FORM_CONCEPT_ID) == 0) {
+					
+					updatePatientAttributes(patient, NEGATIF_CONCEPT_VALUE_ID, obs.getValueDatetime());
+					break;
+				}
```

Comment:
We shouldn't hard code the concept Id here because they are volatile. Pass in the concept uuids and then in the `updatePatientAttributes` method is where you make a query to the concept api to get the concept ID